### DISCUSSION TITLE
Testing deprecateTypingAliases

### DISCUSSION THREAD

**Discussion by mehdigmira** [microsoft/pyright#5848] [General]
Hello,

Tested the new deprecateTypingAliases, but i'm surprised by the output of the following code
```python
# pyright: strict
# pyright: deprecateTypingAliases=true

from __future__ import annotations

# from collections import Counter
import collections

collections.Counter()
```

pyright reports that Counter is deprecated and that I should be using collections.Counter instead, which I'm doing !
Is is a bug ? Or am I misunderstanding this flag ?

**Comment by erictraut** [comment]
This is a bug. Thanks for reporting it.

Implementing this feature was complicated by two factors: 1) Some of the deprecated symbols are named differently from their aliases (like `Tuple` vs `tuple`), but others have the same name (like `Counter`), and 2) Some of the deprecated symbols are simply re-exports of the non-deprecated versions, but some of the others are defined separately in `typing`. This means the test for deprecation needs to be based not only on the full symbol name but also the way in which that symbol was imported into your module. These various cases need to be handled differently.

There are 40 deprecated symbols total, and I missed two cases (`Counter` and `ChainMap`) that require different handling. This will be addressed in the next release.

**Reply by mehdigmira** [reply]
Okay, this flag basically addresses the same issues that ruff or pyupgrade tools already address ? We use ruff for such things because it's has a neat autofix feature for stuff like this, and also allows upgrading python versions easily. I Wonder if there is some benefit in enabling this pyright flag.
